### ISSUE TITLE
Update of letsencrypt certificates not possible

### ISSUE THREAD

**Issue by oli-ver** [Mailu/Mailu#1164]
I cannot update the mailserver's letsencrypt certificates. The following error message is shown in the logs:
```
front_1      | Failed authorization procedure. webmail.myhost.com (http-01): urn:acme:error:connection :: The server could not connect to the client to verify the domain :: Fetching https://127.0.0.1/.well-known/acme-challenge/<the-challenge>: Invalid host in redirect target "127.0.0.1". Only domain names are supported, not IP addresses
```

I don't know where this comes from as in the .env file host names and urls seem to be set correctly. 127.0.0.1 is not part of any configuration parameter.

Mailu-Version: 1.7 (actually the docker image is still listed as 1.6 when executing `docker container ls`, so I am not 100% sure. I did a `docker-compose pull` and `docker-compose up -d` after the 1.7 release.

**Comment by oli-ver** [comment]
Actually the docker-compose.yml has entries for 127.0.0.1 (mailu is executed behind a nginx proxy, but the certificates for letsencrypt are configured to be managed by mailu).
```
...
 # Core services
  front:
    image: ${DOCKER_ORG:-mailu}/${DOCKER_PREFIX:-}nginx:${MAILU_VERSION:-1.6}
    restart: always
    env_file: mailu.env
    logging:
      driver: json-file
    ports:
      - "127.0.0.1:83:80"
      - "127.0.0.1:444:443"
...
```

**Comment by oli-ver** [comment]
It seems to be something not working for the ACME challenge. Normally for websites I usually manage with froxlor the acme-challenge is not forwarded to https. But (if I understand that correctly) the container (or something I configured wrong) seems to try to forward the request to https://127.0.0.1:
```
curl -v http://webmail.myhost.com/.well-known/acme-challenge/something-written-here
*   Trying XX.XXX.XXX.XX...
* TCP_NODELAY set
* Connected to webmail.myhost.com (XX.XXX.XXX.XX) port 80 (#0)
> GET /.well-known/acme-challenge/something-written-here HTTP/1.1
> Host: webmail.myhost.com
> User-Agent: curl/7.52.1
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Server: nginx/1.10.3
< Date: Fri, 06 Sep 2019 11:45:02 GMT
< Content-Type: text/html
< Content-Length: 178
< Connection: keep-alive
< Location: https://127.0.0.1/.well-known/acme-challenge/something-written-here
< Strict-Transport-Security: max-age=31536000
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-Permitted-Cross-Domain-Policies: none
< X-XSS-Protection: 1; mode=block
< Referrer-Policy: same-origin
<
<html>
<head><title>301 Moved Permanently</title></head>
<body bgcolor="white">
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
* Curl_http_done: called premature == 0
* Connection #0 to host webmail.myhost.com left intact
```

**Comment by kaiyou** [comment]
It looks like you included `127.0.0.1` in your hostnames variable, that should only contain hostnames actually accepted by the server, not addresses to listen on.

**Comment by oli-ver** [comment]
@kaiyou Can you please point out which variable you mean? I found out that the IP address set as IP/Port configuration in docker-compose.yml is used for the letsencrypt request. I tried to bind the web component to localhost/127.0.0.1 because I am using mailu behind a nginx webserver installed on the docker host system. As Docker also creates iptables entry these ports are public and I did not want that.

Which variable has to be set to my hostname in order for this to work correctly?


`front_1      | Failed authorization procedure. webmail.myhost.com (http-01): urn:acme:error:connection :: The server could not connect to the client to verify the domain :: Fetching https://MY.IP.ADD.RESS/.well-known/acme-challenge/<hash>: Invalid host in redirect target "MY.IP.ADD.RESS". Only domain names are supported, not IP addresses`

In my mailu.env I do not use 127.0.0.1 at all:

```
# Main mail domain
DOMAIN=myhost.com

# Hostnames for this server, separated with comas
HOSTNAMES=webmail.myhost.com

```

**Comment by oli-ver** [comment]
@kaiyou I ended up setting up letsencrypt externally using froxlor. I noticed something perhaps explaining the issues I had. It seems it had no effect setting `TLS_FLAVOR=cert` instead of `TLS_FLAVOR=letsencrypt`. Mailu was still trying to renew the letsencrypt certificates. Only after deleting the directory `mailu/certs/letsencrypt` folder this stopped.

I assume this is also the reason why the config changes did not help when trying to get letsencrypt to work.

I could not find anything about this behavior in the documentation. I will create a pull request for an update of the docs.
